bootstrap/src/main/java/io/airlift/bootstrap/LifeCycleManager.java

```diff
@@ -39,6 +39,7 @@ public final class LifeCycleManager
 {
     private final Logger log = Logger.get(LifeCycleManager.class);
     private final AtomicReference<State> state = new AtomicReference<>(State.LATENT);
+    private final ConcurrentWeakIdentitySet startedInstances = new ConcurrentWeakIdentitySet();
```

Comment:
isn't that enough:
`private final Set<Object> startedInstances = newSetFromMap(synchronizedMap(new WeakHashMap<>()));` ?

Reply:
would work as long as you put "identity based" wrappers there. Not original objects.

Reply:
I considered that but this was equivalent and simpler.